macros/datediff.sql

```diff
@@ -0,0 +1,43 @@
+{%- macro datediff(part, left_val, right_val, date_format='%Y-%m-%d') -%}
+    {# determines the delta (in `part` units) between first_val and second_val.
+       *Note* the order of left_val, right_val is reversed from Snowflake.
+       ARGS:
+         - part (string) one of 'day', 'week', 'month', 'year'
+         - left_val (date/timestamp) the value before the minus in the equation "left - right"
+         - right_val (date/timestamp) the value after the minus in the equation "left - right"
+         - date_format (pattern) a string pattern for the provided arguments (primarily for BigQuery)
+       RETURNS: An integer representing the delta in `part` units
+    #}
+
+    {%if part not in ('day','week','month','year',) %}
+        {{exceptions.raise_compiler_error("macro datediff for target does not support date part value " ~ part)}}
+    {%- endif -%}
+
+    {%- if target.type ==  'postgres' -%} 
+        {%- if part == 'year' -%}
```

Comment:
any need for a part|lower here?

Reply:
per your comment above I updated it to lower the parts arg across the board - so it's nice and dry now :the_horns: